internal/storage/metastorage/firestore/block_storage.go

```diff
@@ -98,76 +87,23 @@ func (b *blockStorageImpl) GetBlocksByHeightRange(ctx context.Context, tag uint3
 		return nil, err
 	}
 	return b.instrumentGetBlocksByHeightRange.Instrument(ctx, func(ctx context.Context) ([]*chainstorage.BlockMetadata, error) {
-		databasePath := fmt.Sprintf("projects/%s/databases/(default)", b.projectId)
-		parent := databasePath + "/documents"
-		md, ok := metadata.FromOutgoingContext(ctx)
-		if !ok {
-			md = metadata.New(nil)
-		}
-		md = md.Copy()
-		md["google-cloud-resource-prefix"] = []string{databasePath}
-		c, err := b.grpc_client.RunQuery(metadata.NewOutgoingContext(ctx, md), &firestorepb.RunQueryRequest{
-			Parent: parent,
-			QueryType: &firestorepb.RunQueryRequest_StructuredQuery{
-				StructuredQuery: &firestorepb.StructuredQuery{
-					From: []*firestorepb.StructuredQuery_CollectionSelector{
-						{
-							AllDescendants: true,
-						},
-					},
-					StartAt: &firestorepb.Cursor{
-						Values: []*firestorepb.Value{
-							{
-								ValueType: &firestorepb.Value_ReferenceValue{
-									ReferenceValue: b.getCanonicalBlockDocRef(tag, startHeight).Path,
-								},
-							},
-						},
-						Before: true,
-					},
-					EndAt: &firestorepb.Cursor{
-						Values: []*firestorepb.Value{
-							{
-								ValueType: &firestorepb.Value_ReferenceValue{
-									ReferenceValue: b.getCanonicalBlockDocRef(tag, endHeight).Path,
-								},
-							},
-						},
-						Before: true,
-					},
-					OrderBy: []*firestorepb.StructuredQuery_Order{
-						{
-							Field: &firestorepb.StructuredQuery_FieldReference{
-								FieldPath: "__name__",
-							},
-							Direction: firestorepb.StructuredQuery_ASCENDING,
-						},
-					},
-				},
-			},
-		})
-		if err != nil {
-			return nil, xerrors.Errorf("failed to get blocks: %w", err)
-		}
+		startDocRef := b.getCanonicalBlockDocRef(tag, startHeight)
```

Comment:
GetAll uses BatchGetDocuments with document ids: https://github.com/googleapis/google-cloud-go/blob/f2b13307a85e81e278476ea51a359cbb3974667a/firestore/client.go#L253-L269

While using startAt and endAt is a range query, which should be much efficient.

But the cost is the same, it's billed by documents read.

Reply:
I think batch get would be more efficient than range query with orderby on the collection.  Also, we'll have to worry about the padding of the height in sort order on server side.

Reply:
block heights are pad with 0s, it's properly designed for range query, which should theoretically be more efficient because it reduce internal requests to backend storage